torch/csrc/jit/runtime/decomposition_registry.cpp

```diff
@@ -160,6 +161,46 @@ void RegisterDecomposition(
   schema_to_decomposition[&schema] = g;
 }
 
+struct JitDecomp final : torch::autograd::impl::JitDecompInterface {
+  bool has_jit_decomposition_(const c10::FunctionSchema& schema) const override;
+  void run_jit_decomposition_(
+      const c10::OperatorHandle& op,
+      torch::jit::Stack* stack) const override;
```

Comment:
Why do these functions end in '_'?

Reply:
If the member function were also named `run_jit_decomposition` that would hide names outside of the class scope.

Alternatively, I can specify the function with its namespace or use a using declaration.

Reply:
Updated to specify the namespace to get around name hiding